# 🪵🍁🍂🍄🍗🦃🏈 Big Thanksgiving Dinner Analysis 🏈🦃🍗🍄🍂🍁🪵
**Name:**  Camila  Hernandez   <br>
**Class:** Foundation of Data Analytics

In [11]:
# Libraries


In [2]:
import numpy as np
import pandas as pd

In [3]:


# Synthetic Thanksgiving Data
np.random.seed(2310)    # set seed to 2310
n = 50    # set n to 50

# creating the variables first (Age, Dinner, Dessert, Duration)
Age = (np.random.normal(37, 18, n).round(0).clip(5, 90))
Num_Dinner_Servings = np.round(4 - 0.0009 * (Age - 40)**2 + np.random.normal(0, 0.3, n)).clip(1, 5)
Num_Dessert_Servings = np.round(np.clip(2.5 - (Age / 50) + np.random.normal(0, 0.5, n), 0, 3))
Dinner_Duration_Hours =  np.round(1.5 + (Age / 40) + np.random.normal(0, 0.5, n), 1).clip(1, 5)

# creating boolean for whether or not they watched tv (older and younger people more likely to watch)
prob_parade = np.clip(    0.3 + 0.4 * np.exp(-((Age - 15)/20)**2) + 0.4 * np.exp(-((Age - 70)/20)**2),0, 1)
Watched_Parade = np.random.binomial(1, prob_parade)

# creating the data frame using the variables we just made
Thanksgiving = pd.DataFrame({
    'Ages' : Age,
    'Dinner_Servings' : Num_Dinner_Servings,
    'Dessert_Servings' : Num_Dessert_Servings,
    'Duration' : Dinner_Duration_Hours,
    'Macy_Parade' : Watched_Parade
})

Thanksgiving.head()


Ages  Dinner_Servings  Dessert_Servings  Duration  Macy_Parade
0  33.0              4.0               1.0       2.0            1
1  36.0              4.0               2.0       2.2            0
2  37.0              4.0               2.0       3.0            0
3  31.0              4.0               2.0       2.2            1
4  34.0              5.0               2.0       2.0            1

In [4]:
# Use the `describe` function to get the basic summary statistics
Thanksgiving.describe(include='all')


Ages  Dinner_Servings  Dessert_Servings  Duration  Macy_Parade
count  50.000000        50.000000         50.000000  50.00000    50.000000
mean   39.520000         3.920000          1.500000   2.50400     0.680000
std    14.816207         0.395897          0.646813   0.56531     0.471212
min     9.000000         3.000000          0.000000   1.00000     0.000000
25%    31.250000         4.000000          1.000000   2.20000     0.000000
50%    37.500000         4.000000          1.500000   2.55000     1.000000
75%    50.750000         4.000000          2.000000   2.90000     1.000000
max    72.000000         5.000000          3.000000   3.50000     1.000000

## Instructions:
Use this Thanksgiving dataset to explore patterns in the family dining habits.
You’ll need to _**interpret**_ the data using summary statistics, **not just report them**.

You will being using **only the describe function** to answer your questions

### Questions:

1. Looking at the minimum and maximum values of Age, what do they suggest about who attended this Thanksgiving dinner? <br>
**Answer:**  People ranging from ages 9 to 72, this could be a thanksgiving dinner that involves the extended family, multiple generations, multiple ages<br>

2. Looking at the mean and standard deviation of Dinner_Duration, what do they show about how consistent the mealtime was across family members? <br>
**Answer:** <br>the deviation is 0.5 hour meaning most family members spent around the same time on dinner with maybr 30 minutes in difference.

3. If someone stayed for 4.8 hours, would you consider them an outlier? _Use the mean and standard deviation to support your answer (Hint: $mean + 2*std$ OR $IQR=Q3−Q1$ and $Upper Bound=Q3+1.5×IQR$)._ <br>
**Answer:** <br> Using the formula that would equal around 3.63 hours. 4.8 is over this so it would be an outlier.

4. Compare the mean and median of Dinner_Servings. What could a large difference between them indicate about family member behavior? _(Hint: 50% is half way in the data set)_ <br>
**Answer:** <br> mean and median are pretty much the same, meaning most people ate the same amount of food. If there was a bigger difference it would mean some people ate a lot less or more than others.

5. Looking at the max value of Dessert_Servings, what type of guest do you imagine that might be and what could this tell us about dessert consumption overall?  <br>
**Answer:** <br> Since the max amount of dessert was 3, maybe these are poeple that really enjoy sweets or maybe its the children in the family. This tells us that all most all memebers moderate themselves and most have the same number of servings.

6. Suppose the standard deviation of Dinner_Serving is small. What does that suggest about how much everyone ate?  <br>
**Answer:** <br> small deviations mean that most servings were the same. It suggest most people ate the same amount or around the same.

7. Which statistic (min, max, mean, std) best helps you find the “light eaters”? Explain how you would identify them and if the minimum makes sense.<br>
**Answer:** <br> We could use the min to check the smallest values, and them compare this usigng mean, std, and the formula to check if this minimum would make sense or if its just a single person.

8. The range (max − min) of Dinner_Duration is quite large. What could that imply about how the dinner was organized?  <br>
**Answer:** <br>This means there was people that either stayed much longer or much less, this could mean the dinner really didnt have a finish time, people left whenever they wanted eventhough most left at the same time.

9. You suspect that older guests tend to eat fewer desserts. Could you use only the `describe` function to explore whether that idea might be true before doing a full analysis or would we have to include something else?  <br>
**Answer:** <br> The describe option only inncludes statistics of individual columns like mean, min, max, and std, but it cant really show specific information. You would have to filter it by age in some way or make a chart to see comparisons.

10. What does the mean tell you about how common it was for people to watch the Macy’s Day Parade?  <br>
**Answer:** <br> The mean was 0.68 meaning 68% of people watch the parade which is more than half of family members.

11. what does the standard deviation suggest about the variation in watching behavior? Was parade-watching behavior fairly consistent across people, or was there a lot of difference between individuals?  <br>
**Answer:** <br> The std is around 0.47 which means eventhough some people did watch it many didnt so it wasnt completly consistent, there was some variation.

12. In one sentence, summarize what kind of Thanksgiving this dataset represents. Focus on what the numbers imply rather than just stating the numbers.  <br>
**Answer:** <br> This represents a big family Thanksgiving dinner where there are members of many ages that seem to enjoy dessert,  the duration of their dinner was consistent and then most sat and watch the Macy Parade.

13. Suggest some visuals that would be ideal for exploring this data and state their purpose. <br>
**Answer:** <br> Boxplots and Histograms would be the easiest ways to analyze the data. Histograms these display in simple ways how data is distributed this could be age or dinner duration. Boxplots can be used to show medians mins and maxes.
